contracts/adapters/AutoRepay.sol

```diff
@@ -0,0 +1,313 @@
+    {
+      IERC20(repayParams.debtAsset).safeApprove(address(LENDING_POOL), 0);
+      IERC20(repayParams.debtAsset).safeApprove(address(LENDING_POOL), repayParams.debtRepayAmount);
+      uint256 repaidAmount = IERC20(repayParams.debtAsset).balanceOf(address(this));
+      LENDING_POOL.repay(
+        repayParams.debtAsset,
+        repayParams.debtRepayAmount,
+        repayParams.rateMode,
+        repayParams.user
+      );
+      repaidAmount = repaidAmount.sub(IERC20(repayParams.debtAsset).balanceOf(address(this)));
+
+      uint256 maxCollateralToSwap = repayParams.collateralAmount;
+      if (repaidAmount < repayParams.debtRepayAmount) {
+        maxCollateralToSwap = maxCollateralToSwap.mul(repaidAmount).div(
+          repayParams.debtRepayAmount
+        );
+      }
+
+      repayParams.collateralAmount = maxCollateralToSwap;
+      repayParams.debtRepayAmount = repaidAmount;
+    }
+
+    _doSwapAndPullWithFee(repayParams, permitSignature, caller, premiums[0]);
+
+    // Repay flashloan. Approves for 0 first to comply with tokens that implement the anti frontrunning approval fix.
+    IERC20(repayParams.debtAsset).safeApprove(address(LENDING_POOL), 0);
+    IERC20(repayParams.debtAsset).safeApprove(address(LENDING_POOL), amounts[0].add(premiums[0]));
+
+    return true;
+  }
+
+  function increaseHealthFactor(
+    RepayParams memory repayParams,
+    PermitSignature calldata permitSignature
+  ) public {
+    require(isWhitelisted(msg.sender), 'Caller is not whitelisted');
+    _checkMinHealthFactor(repayParams.user);
+    if (repayParams.useFlashloan) {
+      bytes memory params = abi.encode(repayParams, permitSignature, msg.sender);
+      address[] memory assets = new address[](1);
+      assets[0] = repayParams.debtAsset;
+      uint256[] memory amounts = new uint256[](1);
+      amounts[0] = repayParams.debtRepayAmount;
+      uint256[] memory modes = new uint256[](1);
+      modes[0] = 0;
+      LENDING_POOL.flashLoan(address(this), assets, amounts, modes, repayParams.user, params, 0);
+    } else {
+      DataTypes.ReserveData memory debtReserveData = _getReserveData(repayParams.debtAsset);
+      uint256 amountToRepay;
+      {
+        address debtToken = DataTypes.InterestRateMode(repayParams.rateMode) ==
+          DataTypes.InterestRateMode.STABLE
+          ? debtReserveData.stableDebtTokenAddress
+          : debtReserveData.variableDebtTokenAddress;
+        uint256 currentDebt = IERC20(debtToken).balanceOf(repayParams.user);
+        amountToRepay = repayParams.debtRepayAmount <= currentDebt
+          ? repayParams.debtRepayAmount
+          : currentDebt;
+      }
+      uint256 maxCollateralToSwap = repayParams.collateralAmount;
+      if (amountToRepay < repayParams.debtRepayAmount) {
+        maxCollateralToSwap = maxCollateralToSwap.mul(amountToRepay).div(
+          repayParams.debtRepayAmount
+        );
+      }
+      repayParams.collateralAmount = maxCollateralToSwap;
```

Comment:
```suggestion
      if (amountToRepay < repayParams.debtRepayAmount) {
        repayParams.collateralAmount = maxCollateralToSwap.mul(amountToRepay).div(
          repayParams.debtRepayAmount
        );
      }
```

Reply:
again, no need to introduce `maxCollateralToSwap` at all.